src/main/java/appeng/core/AELog.java

```diff
@@ -19,93 +19,208 @@
 package appeng.core;
 
 
-import org.apache.logging.log4j.Level;
+import javax.annotation.Nonnull;
 
-import cpw.mods.fml.relauncher.FMLRelaunchLog;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
 
 import appeng.core.features.AEFeature;
 import appeng.tile.AEBaseTile;
 import appeng.util.Platform;
 
 
+/**
+ * Utility class for easier logging.
+ */
 public final class AELog
 {
 
-	public static final FMLRelaunchLog INSTANCE = FMLRelaunchLog.log;
+	private static final String LOGGER_PREFIX = "AE2:";
+	private static final String SERVER_SUFFIX = "S";
+	private static final String CLIENT_SUFFIX = "C";
 
 	private AELog()
 	{
 	}
 
-	public static void warning( final String format, final Object... data )
+	/**
+	 * Returns a {@link Logger} logger suitable for the effective side (client/server).
+	 *
+	 * @return a suitable logger instance
+	 */
+	private static Logger getLogger()
 	{
-		log( Level.WARN, format, data );
+		final String loggerName = LOGGER_PREFIX + ( Platform.isServer() ? SERVER_SUFFIX : CLIENT_SUFFIX );
+		final Logger logger = LogManager.getLogger( loggerName );
+
+		return logger;
 	}
 
-	private static void log( final Level level, final String format, final Object... data )
+	/**
+	 * Logs a formatted message with a specific log level.
+	 *
+	 * This uses {@link String#format(String, Object...)} as opposed to the {@link ParameterizedMessage} to allow a more
+	 * flexible formatting.
+	 *
+	 * The output can be globally disabled via the configuration file.
+	 *
+	 * @param level the intended level.
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Level level, @Nonnull final String message, final Object... params )
 	{
 		if( AEConfig.instance == null || AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
 		{
-			FMLRelaunchLog.log( "AE2:" + ( Platform.isServer() ? "S" : "C" ), level, format, data );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( level, formattedMessage );
 		}
 	}
 
-	public static void grinder( final String o )
+	/**
+	 * Log an exception with a custom message formated via {@link String#format(String, Object...)}
+	 *
+	 * Similar to {@link AELog#log(Level, String, Object...)}.
+	 *
+	 * @see AELog#log(Level, String, Object...)
+	 *
+	 * @param exception
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Throwable exception, @Nonnull String message, final Object... params )
```

Comment:
Currently I want to keep the amount of methods as small as possible. It is probably fine to change `WARN` to `ERROR` here and should the need arise for a different loglevel, then extend it accordingly. 

I really do not want to let `AELog` implement `Logger` with like hundred different methods and proxy everything. Just because we can have a more fine grained logging, because it is not possible to do this via forge.